src/machine/machine_rp2_timer.go

```diff
@@ -101,3 +101,15 @@ func (tmr *timerType) lightSleep(us uint64) {
 	// Disable interrupt
 	intr.Disable()
 }
+
+// setDbgPause sets whether this timer is paused when a debugger is connected.
+func (tmr *timerType) setDbgPause(enable bool) {
+	const bitPos = 1
+	const bitMask = 0b11
+	val := uint32(0b00)
+	if enable {
+		// Disable timer when debugger is connected to either core.
+		val = 0b11
+	}
+	tmr.dbgPause.ReplaceBits(val, bitMask, bitPos)
```

Comment:
Nit: use constants from package `rp`:

```go
val := uint32(0)
mask := rp.WATCHDOG_CTRL_PAUSE_DBG0_Msk|
    rp.WATCHDOG_CTRL_PAUSE_DBG1_Msk
if enable {
    val = mask
}
tmr.dbgPause.ReplaceBits(val, mask, 0)
```

(untested)

Reply:
Yeah, I'd prefer to use a mask constant, but that's for a different register with the same name unfortunately. They also switched from one timer on the RP2040 to two on RP2350 (one for secure code, one for non-secure) so the naming switched (TIMER_DBGPAUSE_DBG[01]_Msk vs TIMER[01]_DBGPAUSE_DBG[01]_Msk)